mesh/mesh.cpp

```diff
@@ -1302,19 +1380,60 @@ void Mesh::AddHexAsWedges(const int *vi, int attr)
    }
 }
 
-void Mesh::AddBdrSegment(const int *vi, int attr)
+int Mesh::AddElement(Element *elem)
+{
+   CheckEnlarge(boundary, NumOfBdrElements);
```

Comment:
@jakubcerveny Is this line intended? Should `CheckEnlarge` really be called in `AddElement`? This has broken a code I have, which uses a simple way to construct a mesh by just calling `AddVertex` and `AddElement` for a set of vertices and elements in another mesh. There are no boundary elements, so `FinalizeCheck()` fails due to `boundary.Size() == 1` not being equal to `NumOfBdrElements == 0`, which is caused by this `CheckEnlarge` (when `boundary` has size 0, it gets set to size 1).

Reply:
Oops, this is a typo (or a copy/paste error), thanks for pointing this out. @tzanio can I push a one-line fix directly into master?

Reply:
Maybe that's not such a good idea. I pushed the fix instead to this branch (nc-init-dev), but I'm not sure how to revive this PR...

Reply:
I made PR #1764 for this one-line fix and a unit test that should prevent this (unusual) method of mesh construction from failing in the future.